src/django_components/middleware.py

```diff
@@ -33,14 +36,32 @@ class ComponentDependencyMiddleware:
     def __init__(self, get_response: "Callable[[HttpRequest], HttpResponse]") -> None:
         self.get_response = get_response
 
+        if iscoroutinefunction(self.get_response):
+            markcoroutinefunction(self)
+
     def __call__(self, request: HttpRequest) -> HttpResponseBase:
+
+        if iscoroutinefunction(self):
+            return self.__acall__(request)
+
         response = self.get_response(request)
+        response = self.process_response(request)
```

Comment:
Ah, maybe the issue is that we pass `request` here instead of `response`?

Reply:
That'd do it :) Hopefully it's working with that change